message_ix/core.py

```diff
@@ -59,10 +71,58 @@ def __init__(
         # Scheme returned by database
         assert self.scheme == "MESSAGE", self.scheme
 
+        if sys.version_info >= (3, 10):
+            from ixmp.backend.ixmp4 import IXMP4Backend
+
+            from message_ix.util.scenario_setup import add_default_data
+
+            if isinstance(self.platform._backend, IXMP4Backend):
```

Comment:
In general here I'd favour (and will add) a utility function that collapses these two conditionals into one.